src/messaging/ExchangeMgr.cpp

```diff
@@ -106,11 +115,13 @@ ExchangeContext * ExchangeManager::NewContext(const NodeId & peerNodeId, Exchang
 
 ExchangeContext * ExchangeManager::FindContext(NodeId peerNodeId, ExchangeDelegate * delegate, bool isInitiator)
 {
-    for (auto & ec : ContextPool)
+    ExchangeContext * ec = mContextPool;
+
+    for (int i = 0; i < CHIP_CONFIG_MAX_EXCHANGE_CONTEXTS; i++, ec++)
```

Comment:
Why this change away from a ranged for loop to something more confusing and error-prone?